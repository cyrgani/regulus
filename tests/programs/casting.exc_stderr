TypeError: Unable to cast a to int
at programs/casting.re:10:35
at programs/casting.re:10:31

