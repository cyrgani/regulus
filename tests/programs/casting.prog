_(
    # allowed
    assert(==("2", string(2))),
    assert(==("2", string("2"))),
    assert(==(2, int("2"))),
    assert(==(2, int(string(2)))),
    assert(==(true, bool(1))),
    assert(==(string(int(bool(int("2")))), "1")),
)