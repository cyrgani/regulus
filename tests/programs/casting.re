# allowed
assert_eq("2", string(2)),
assert_eq("2", string("2")),
assert_eq(2, int("2")),
assert_eq(2, int(string(2))),
assert_eq(true, bool(1)),
assert_eq(string(int(bool(int("2")))), "1"),

# disallowed
__builtin_print_catch(int("a")),
