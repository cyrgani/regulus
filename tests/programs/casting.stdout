TypeError: Unable to cast a to int
at programs/casting.re:10:26
at programs/casting.re:10:22
